framework/grid/DataColumn.php

```diff
@@ -189,6 +189,12 @@ protected function renderFilterCellContent()
             if (is_array($this->filter)) {
                 $options = array_merge(['prompt' => ''], $this->filterInputOptions);
                 return Html::activeDropDownList($model, $this->attribute, $this->filter, $options) . $error;
+            } elseif ($this->format === 'boolean') {
+                $options = array_merge(['prompt' => ''], $this->filterInputOptions);
+                return Html::activeDropDownList($model, $this->attribute, [
+                        $this->grid->formatter->booleanFormat[0],
+                        $this->grid->formatter->booleanFormat[1]
```

Comment:
Also I'm not sure about framework's code style about this but with arrays I prefer to add comma after the last element:

```php
$this->grid->formatter->booleanFormat[1],
```

So ignore this if it's not crucial.